src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/internalLinks/index.js

```diff
@@ -21,82 +20,76 @@ to the same schema used for external links
-const buildUrlString = ({ url, linkText }) => {
-  // simorgh#5078 - The psammead-rich-text-transforms package does not like chevrons
-  const parsedLinkText = linkText.replace('<', '%3C').replace('>', '%3E');
-  return `
-        <link>
-            <caption>${parsedLinkText}</caption>
-            <altText>${parsedLinkText}</altText>
-            <url href="${url}" platform="highweb"/>
-            <url href="${url}" platform="enhancedmobile"/>
-        </link>
-    `.replace(/(\r?\n|\r)\s*/g, '');
-};
+const getUrl = path(['locators', 'href']);
 
-// Given a block, an <itemMeta> match, and the corresponding metadata,
-// replace the match with the full link XML
-const provideLinkXML = ({ block, match, metadata }) => {
-  if (!metadata) return;
+const getLinkText = path(['headlines', 'overtyped']);
 
-  const url = path(['locators', 'href'], metadata);
-  const linkText = path(['headlines', 'overtyped'], metadata);
+const itemIdEndsWith = string => ({ id }) => id.endsWith(string);
 
-  if (!url || !linkText) return;
+const getItemMetadata = (id, metadata) => metadata.find(itemIdEndsWith(id));
 
-  // eslint-disable-next-line no-param-reassign
-  block.text = block.text.replace(
-    match,
-    buildUrlString({
-      url,
-      linkText,
-    }),
-  );
-};
+// Given a block that has text, gather up any <itemMeta> tags within that text
+const getItemMetaMatches = text =>
+  Array.from(text.matchAll(/<itemMeta>(.*?)<\/itemMeta>/g));
 
-// Given an array of metadata and the id stored within the <itemMeta> tag,
-// find the corresponding metadata object from that array
-const findMetadata = (metadata, id) => {
-  return metadata.find(item => {
-    return item.id.endsWith(id);
-  });
-};
+const whiteSpaceRegex = /(\r?\n|\r)\s*/g;
 
-// Given a block that has text, gather up any <itemMeta> tags within that text
-const parseText = block => {
-  const matches = Array.from(
-    block.text.matchAll(/<itemMeta>(.*?)<\/itemMeta>/g),
-  );
-  if (!matches.length) return;
-
-  // Iterate over every <itemMeta> tag, and pass it off to be replaced
-  matches.forEach(([match, id]) => {
-    provideLinkXML({ block, match, metadata: findMetadata(block.meta, id) });
-  });
-};
+// This is the template CPS uses for external links - we already support these
+// A later transformer converts them into the optimo format
 
-// Recursively iterate through all blocks - if a block has a "text" property, send it off for processing
-const findAndEnrichTextValues = blocks => {
-  if (!Array.isArray(blocks)) return;
-  blocks.forEach(block => {
-    if (block.text) {
-      parseText(block);
+const getLinkXML = (url, text) =>
+  `
+<link>
+    <caption>${text}</caption>
+    <altText>${text}</altText>
+    <url href="${url}" platform="highweb"/>
+    <url href="${url}" platform="enhancedmobile"/>
+</link>
+`.replace(whiteSpaceRegex, '');
+
+const replaceChevrons = text => text.replace('<', '%3C').replace('>', '%3E');
```

Comment:
Needs changed to this following the update to fix the chevron issue: https://github.com/bbc/simorgh/pull/5143/files

(also has updated unit tests in that PR)

```suggestion
const replaceChevrons = text => text.replace('<', '&lt;').replace('>', '&gt;');
```